lectures/tools_and_techniques/ar1_processes.md

```diff
@@ -0,0 +1,575 @@
+partly because they help us understand important concepts.
+
+```{code-cell} julia
+using LinearAlgebra, Statistics
+using Distributions, LaTeXStrings, Plots
+```
+
+## The AR(1) Model
+
+The **AR(1) model** (autoregressive model of order 1) takes the form
+
+```{math}
+:label: can_ar1
+
+X_{t+1} = a X_t + b + c W_{t+1}
+```
+
+where $a, b, c$ are scalar-valued parameters.
+
+This law of motion generates a time series $\{ X_t\}$ as soon as we
+specify an initial condition $X_0$.
+
+This is called the **state process** and the state space is $\mathbb R$.
+
+To make things even simpler, we will assume that
+
+* the process $\{ W_t \}$ is IID and standard normal,
+* the initial condition $X_0$ is drawn from the normal distribution $N(\mu_0, v_0)$ and
+* the initial condition $X_0$ is independent of $\{ W_t \}$.
+
+### Moving Average Representation
+
+Iterating backwards from time $t$, we obtain
+
+$$
+X_t = a X_{t-1} + b +  c W_t
+        = a^2 X_{t-2} + a b + a c W_{t-1} + b + c W_t
+        = \cdots
+$$
+
+If we work all the way back to time zero, we get
+
+```{math}
+:label: ar1_ma
+
+X_t = a^t X_0 + b \sum_{j=0}^{t-1} a^j +
+        c \sum_{j=0}^{t-1} a^j  W_{t-j}
+```
+
+Equation {eq}`ar1_ma` shows that $X_t$ is a well defined random variable, the value of which depends on
+
+* the parameters,
+* the initial condition $X_0$ and
+* the shocks $W_1, \ldots W_t$ from time $t=1$ to the present.
+
+Throughout, the symbol $\psi_t$ will be used to refer to the
+density of this random variable $X_t$.
+
+### Distribution Dynamics
+
+One of the nice things about this model is that it's so easy to trace out the sequence of distributions $\{ \psi_t \}$ corresponding to the time
+series $\{ X_t\}$.
+
+To see this, we first note that $X_t$ is normally distributed for each $t$.
+
+This is immediate from {eq}`ar1_ma`, since linear combinations of independent
+normal random variables are normal.
+
+Given that $X_t$ is normally distributed, we will know the full distribution
+$\psi_t$ if we can pin down its first two moments.
+
+Let $\mu_t$ and $v_t$ denote the mean and variance
+of $X_t$ respectively.
+
+We can pin down these values from {eq}`ar1_ma` or we can use the following
+recursive expressions:
+
+```{math}
+:label: dyn_tm
+
+\mu_{t+1} = a \mu_t + b
+\quad \text{and} \quad
+v_{t+1} = a^2 v_t + c^2
+```
+
+These expressions are obtained from {eq}`can_ar1` by taking, respectively, the expectation and variance of both sides of the equality.
+
+In calculating the second expression, we are using the fact that $X_t$
+and $W_{t+1}$ are independent.
+
+(This follows from our assumptions and {eq}`ar1_ma`.)
+
+Given the dynamics in {eq}`ar1_ma` and initial conditions $\mu_0,
+v_0$, we obtain $\mu_t, v_t$ and hence
+
+$$
+\psi_t = N(\mu_t, v_t)
+$$
+
+The following code uses these facts to track the sequence of marginal
+distributions $\{ \psi_t \}$.
+
+The parameters are
+
+```{code-cell} julia
+a, b, c = 0.9, 0.1, 0.5
```

Comment:
We should never have these on the same line.  It is an enormous source of bugs.  When there are lots of parameters to collect, we should keep them in a named tuple which are not error-prone because the order doesn't change.

For some examples it is probbly a good idea to put things in a named tuple, but it is overkill here.